### ISSUE TITLE
Integrate Nextcloud ?

### ISSUE THREAD

**Issue by Perdouille** [RainLoop/rainloop-webmail#1329]
Hello ! 

I am using [the Nextcloud version of Rainloop](https://apps.nextcloud.com/apps/rainloop), and there is options to integrate Google Drive, Dropbox, ... but not Nextcloud files

Is it possible ?

Thanks !

**Comment by aleks-nl5** [comment]
Also interested in the issue!
Save attachments possible in NC (called truth - 'save to owncloud')

**Comment by DJCrashdummy** [comment]
i would appreciate such a feature!
well... maybe a generic connector to any WebDAV-storage (like the CardDAV-function for the adressbook) will also fit the bill and bring the most benefit.

**Comment by gtrdblt** [comment]
It would be really nice to be able to save/attach files from our own cloud provider :)

**Comment by pierre-alain-b** [comment]
Saving attachments does work. Attachments are saved into an "attachment" folder at the root of the data folder.

Do you know if this works with owncloud? I assume it does not - otherwise I would try to backport the functionality from the owncloud version and make it work with nextcloud.

**Comment by gtrdblt** [comment]
Unfortunately, I don't see any way to save an attachment to the server hard drive :( Is there any options to activate to enable this function ? 

Thank you

**Comment by pierre-alain-b** [comment]
When you have an attachment in the mail, you can make this appear:
![image](https://cloud.githubusercontent.com/assets/386857/24094538/71866ee0-0d59-11e7-8dde-f8178caafc3a.png)

**Comment by gtrdblt** [comment]
Ok now I see it :) I works indeed (with nextcloud) !

Now do you know if it would be possible to browse the server files to attach them to an email ?

Thank you (and thank you for your work also !)

**Comment by pierre-alain-b** [comment]
Well, this is much more challenging. There is a Mail app that has been recently published and seems to do so - so maybe we could get some inspiration from there.
That being said, I expect this will require some heavy changes within Rainloop base code.

If someone is really motivated, I am sure a pull request would be welcome by the Rainloop team!

**Comment by gtrdblt** [comment]
I'd like to, but I'm pretty sure I don't have the skills to do that :) But I'm willing to help testing the functionality some day !

**Comment by akmet** [comment]
This sounds very interesting!
Maybe I have a look at the source of the Mail App after my exams next week :-)

**Comment by mehrdaad** [comment]
@Perdouille did you find a way?

**Comment by mehrdaad** [comment]
hey @pierre-alain-b sorry for noob question, but would you tell me how to achieve this as illustrated in the picture?

**Comment by pierre-alain-b** [comment]
There is a small cog in the bottom right corner of the header box in Rainloop when the message has an attached file

**Comment by mehrdaad** [comment]
Saw it! Is it possible to integrate it with nextcloud? since what you're mentioning is for owncloud(correct me if I'm wrong)

**Comment by mehrdaad** [comment]
@pierre-alain-b correct me if I'm wrong but option of saving to owncloud is available if rainloop is installed as an app in owncloud right?

**Comment by pierre-alain-b** [comment]
yes but this is similar if the app is installed in nextcloud, this appears and works as well

**Comment by Perdouille** [comment]
@mehrdaad Sadly I couldn't find a way to do what I wanted (Attach a file from cloud to mail) :(

**Comment by mehrdaad** [comment]
@pierre-alain-b so what should a developer do if his rainloop instance is not on owncloud/nextcloud.  
Is there any api or sth to do this?

**Comment by pierre-alain-b** [comment]
You mean if the app is standalone? ah well, in such case clearly it will not be automatic. Maybe the simplest would be to develop a plugin for Rainloop to push a file into a owncloud/nextcloud through WebDAV?
https://docs.nextcloud.com/server/12/developer_manual/client_apis/WebDAV/index.html

**Comment by mehrdaad** [comment]
Right. Just to clarify suppose that each account in rainloop has an account in nextcloud.  
Now, save to nextcloud option will save an attachment to that user's storage.  
Also there should be an option to upload from the cloud, which will open the user's storage direction in nextcloud and the user will be able to choose a file.  
Does using webdav supports this use case?  
What would be the authentication process?
Also don't you consider using oath2 instead?
